contributing/README.md

```diff
@@ -0,0 +1,40 @@
+# Contributing
+
+## Table of Contents
+- [BonnyCI Workflow](#bonnyci-workflow)
+- [Commits](#commits)
+  - [Commit Message](#commit-message)
+    - [Sign-off](#sign-off)
+- [Pull Requests](#pull-requests)
+
+## BonnyCI Workflow
+We use the [GitHub flow](https://guides.github.com/introduction/flow/) for code changes.
+
+## Commits
+Changes should include small commits that produce a functional tree. Doing so helps keep the repository history clear and simple to bisectable.
+
+### Commit Message
+Please keep commit messages concise and informative. If the commit relates to an open issue, please include a line in the commit message such as `Fixes #XXX`. When in doubt, follow [this guide](http://chris.beams.io/posts/git-commit/) on writing commit messages.
+
+#### Sign-off
+To improve tracking of who did what, all commits must have a "sign-off". This sign-off is a line at the end of the commit message certifying that you have read and agreed to the [Developers' Agreement](DEVELOPER_AGREEMENT.md). Certify that you agree by adding a sign-off to your commit (**NOTE:** Use your real name, please. No pseudonyms or anonymous contributions excepted.):
+
+```
+Signed-off-by: Random J Developer <random@developer.example.org>
+```
+
+This is doable via the `-s` flag for `git-commit`:
+
+```shell
+$ git add foo
+$ git commit -s -m "Adding foo file"
```

Comment:
That's fair. Since you included the link to writing a good commit message we can let this slide.